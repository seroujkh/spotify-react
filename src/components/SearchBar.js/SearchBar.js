import logo from '../../assets/icons/logo.png';
import PropTypes from 'prop-types';

const SearchBar =( {onTextChangeHandler,value}) => {
    const onSubmit = (event) => {
        event.preventDefault();
    }
    return (
        <div className="container pt-5">
            <div className=' searchbar__container bg-green '>
                <form onSubmit={onSubmit}>
                    <img src={logo} className='logo'  alt="" />
                    <input type='text' className='ml-lg-4 search-value-input'data-testid="search-value-input" onChange={(e) => onTextChangeHandler(e)} value={value!=="..." ? value : ''} />
                    <input type="submit" className="d-none" />
                </form>
            </div>
        </div>
    )
}

SearchBar.propTypes ={
    onTextChangeHandler : PropTypes.func.isRequired,
    value : PropTypes.string
}

export default SearchBar;