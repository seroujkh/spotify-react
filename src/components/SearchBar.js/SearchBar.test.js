import { render, screen, } from '@testing-library/react';
import SearchBar from './SearchBar';
import { shallow } from 'enzyme';


describe('SearchBar', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<SearchBar onTextChangeHandler={jest.fn()} value="" />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());


  it('should render the value of value', () => {
    wrapper.setProps({ value: 'test' });
    expect(wrapper.find('.search-value-input').prop('value')).toEqual('test');
  });

});

describe('Search Bar', () => {

  test('should render the value', () => {
    const { queryByTestId } = render(<SearchBar onTextChangeHandler={jest.fn()} value="test" />);
    const searchInput = queryByTestId("search-value-input");
    expect(searchInput.value).toBe('test');
  });

});