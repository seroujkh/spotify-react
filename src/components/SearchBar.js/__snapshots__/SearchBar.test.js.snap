// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `ShallowWrapper {}`;
